### PR TITLE
templated some mathutils, and used raw_value for some outputs 

### PR DESCRIPTION
**Pull request by tophmatthews**
ref #13926

### CODE REVIEW

File: modules/tensor_mechanics/src/materials/ADSingleVariableReturnMappingSolution.C [hunk 1]
```diff
@@ -393,7 +395,7 @@ ADSingleVariableReturnMappingSolution<compute_stage>::outputIterationStep(
                  << " trial_stress=" << MetaPhysicL::raw_value(effective_trial_stress)
                  << " scalar=" << MetaPhysicL::raw_value(scalar) << " residual=" << residual
                  << " ref_res=" << reference_residual
-                 << " rel_res=" << std::abs(MetaPhysicL::raw_value(residual)) / reference_residual
+                 << " rel_res=" << std::abs(residual) / reference_residual
```
Comment (dschwen) [inline modules/tensor_mechanics/src/materials/ADSingleVariableReturnMappingSolution.C]: why did you remove that raw_value?

**Review by dschwen — APPROVED** [review]
Looks good to me

File: modules/tensor_mechanics/src/materials/ADSingleVariableReturnMappingSolution.C (hunk 1 above)
Comment (tophmatthews) [reply modules/tensor_mechanics/src/materials/ADSingleVariableReturnMappingSolution.C]: It's already converted a couple lines up

**Comment by tophmatthews** [comment]
Any reason not to template a bunch more tensor mechanics things @dschwen ? Any disadvantage?
